### ISSUE TITLE
Using PPA (Trusty) v1.5.5 with websockets not working

### ISSUE THREAD

**Issue by Scotmantartan** [eclipse/mosquitto#1105]
We were using this mosquitto ppa for our Ubuntu 14.04 installations. We were using version 1.4.5 with websockets successfully.

When trying  version 1.5.5 our connection just hangs. Using HiveMQ websockets showcase client hangs and eventually the browser reports it unresponsive and closes it.

I've attached a capture from the moment I connect to when the browser kills the page.

Any ideas?
[packetcapture (2).zip](https://github.com/eclipse/mosquitto/files/2750494/packetcapture.2.zip)

**Comment by ralight** [comment]
Mosquitto 1.5 dropped support for some old versions of libwebsockets, and that includes the (6 year old) 1.2.2 provided in Trusty. It's problematic for the project to distribute a later version of libwebsockets in the PPA. The mosquitto version in the ppa has got patches to try and include libwebsockets support, but it can only go so far.

I'd suggest the best plan for you would be to compile your own version of libwebsockets (I'd suggest version 2.4.2) and mosquitto. If you'd like some help with that let me know.

Trusty is end of life for regular support at the end of April, that will mark the end of support from the mosquitto project as well.

**Comment by Scotmantartan** [comment]
Thanks for the reply Roger.

I expected as much and completely understand. 

I've spent the last couple of days compiling lidbwebsokets to use with the trusty ppa which as you'll know didn't work. Then trying to build deb packages from current Mosquitto and libwebsockets but failed miserably. I finally just compiled and installed both which might have to be the stop gap. It just we have a few servers this will need done to so a package would have been ideal.

We are in the process of upgrading our code to allow a move to 18.04.

Thanks again.
Patrick
